nodes/generators_extended/spiral_mk2.py

```diff
@@ -0,0 +1,655 @@
+        self.inputs.new('SvStringsSocket', "h").prop_name = 'height'
+        self.inputs.new('SvStringsSocket', "p").prop_name = 'phase'
+        self.inputs.new('SvStringsSocket', "a").prop_name = 'arms'
+
+        self.outputs.new('SvVerticesSocket', "Vertices")
+        self.outputs.new('SvStringsSocket', "Edges")
+
+        self.presets = "ARCHIMEDEAN"
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'presets')
+        layout.prop(self, 'sType', text="")
+        col = layout.column(align=True)
+        if self.sType in ("LOGARITHMIC", "ARCHIMEDEAN", "SPIRANGLE"):
+            row = col.row(align=True)
+            row.prop(self, 'normalize', expand=True)
+        row = col.row(align=True)
+        row.prop(self, 'flip', text="Flip", toggle=True)
+        row.prop(self, 'separate', text="Separate", toggle=True)
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_angle_units_buttons(context, layout)
+
+    def process(self):
+        outputs = self.outputs
+        # return if no outputs are connected
+        if not any(s.is_linked for s in outputs):
+            return
+
+        # input values lists (single or multi value)
+        inputs = self.inputs
+        input_R = inputs["R"].sv_get()[0]  # list of exterior radii
+        input_r = inputs["r"].sv_get()[0]  # list of interior radii
+        input_e = inputs["e"].sv_get()[0]  # list of exponents
+        input_t = inputs["t"].sv_get()[0]  # list of turns
+        input_n = inputs["n"].sv_get()[0]  # list of curve resolutions
+        input_s = inputs["s"].sv_get()[0]  # list of scales
+        input_h = inputs["h"].sv_get()[0]  # list of heights (z)
+        input_p = inputs["p"].sv_get()[0]  # list of phases
+        input_a = inputs["a"].sv_get()[0]  # list of arms
+
+        # sanitize the input
+        input_R = list(map(lambda x: max(0.0, x), input_R))
+        input_r = list(map(lambda x: max(0.0, x), input_r))
+        input_t = list(map(lambda x: max(1, int(x)), input_t))
+        input_n = list(map(lambda x: max(3, int(x)), input_n))
+        input_a = list(map(lambda x: max(1, int(x)), input_a))
+
+        # extra parameters
+        f = self.flip  # flip direction
+
+        parameters = match_long_repeat([input_R, input_r, input_e, input_t,
+                                        input_n, input_s, input_h, input_p, input_a])
+
+        # conversion factor from the current angle units to radians
+        au = self.radians_conversion_factor()
+
+        make_spiral = eval("make_" + self.sType.lower() + "_spiral")
+
+        verts_list = []
+        edges_list = []
+        for R, r, e, t, n, s, h, p, a in zip(*parameters):
+            p = p * au
+            arm_verts = []
+            arm_edges = []
+            for i in range(a):  # generate each arm
+                pa = p + 2 * pi / a * i
+                settings = [R, r, e, t, n, s, h, pa, f]  # spiral settings
```

Comment:
I would suggest using named tuple here.

Reply:
Can you give me an example of what you mean?

Reply:
Just as example:
```Python
from typing import NamedTuple

class NodeSettings(NamedTuple):
    vertices: list
    polygons: list
    some_property: float = 0.0

node_settings = NodeSettings([v1, v2, v3], [polygon], 1)

extrude(node_settings.vertices, node_settings.polygons, node_settings.some_property)
```
More information here: https://docs.python.org/3/library/typing.html#typing.NamedTuple